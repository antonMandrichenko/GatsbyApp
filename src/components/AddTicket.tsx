import React, { useContext } from "react"
import { makeStyles } from "@material-ui/core/styles"
import Portal from "@material-ui/core/Portal"
import Button from "@material-ui/core/Button"
import TextField from "@material-ui/core/TextField"
import AppContext from "../context/AppContext"

const useStyles = makeStyles(theme => ({
  button: {
    marginTop: theme.spacing(1),
  },
  alert: {
    padding: theme.spacing(1),
    margin: theme.spacing(1, 0),
    border: "1px solid",
  },
}))

const AddTicket: React.FC = () => {
  const classes = useStyles()
  const [show, setShow] = React.useState(false)
  const [disable, setDisable] = React.useState(false)
  const [description, setText] = React.useState("")
  const container = React.useRef(null)
  const { addedTicketToList, loadingData } = useContext(AppContext)

  const handleClick = () => {
    setShow(!show)
    setDisable(!disable)
  }

  const closeMenu = (description: string) => {
    setShow(false)
    setDisable(false)
    if (description) {
      addedTicketToList({ description })
    }
  }

  return (
    <div>
      <Button
        variant="contained"
        color="secondary"
        className={classes.button}
        fullWidth
        onClick={handleClick}
        disabled={disable || loadingData}
        data-testid="button-first"
      >
        Create list item
      </Button>
      {show ? (
        <Portal container={container.current}>
          <TextField
            id="standard-disabled"
            label="Description"
            onChange={(e: React.ChangeEvent<Element>) => {
              setText(e.target.value)
            }}
            data-testid="text"
            
          />
          <Button
            variant="contained"
            color="secondary"
            className={classes.button}
            fullWidth
            onClick={() => {
              closeMenu(description)
            }}
            data-testid="button-second"
          >
            {!description ? "Close" : "Add item to list"}
          </Button>
        </Portal>
      ) : null}
      <div className={classes.alert} ref={container} />
    </div>
  )
}

export default AddTicket
