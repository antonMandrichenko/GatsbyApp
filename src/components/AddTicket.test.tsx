import React from "react"
import { render, fireEvent } from "@testing-library/react"
import AddTicket from "./AddTicket"


describe("<AddTicket />", () => {
  test("should display text-field", async () => {
    const { getByTestId } = render(<AddTicket />)
    const button = await getByTestId("button-first")
    expect(button).toBeTruthy()
    fireEvent.click(button)
    const textField = await getByTestId("text")
    expect(textField).toBeTruthy()
  })

  test("changes view of button", async () => {
    const { getByTestId } = render(<AddTicket />)
    const button = await getByTestId("button-first")
    expect(button).toBeTruthy()
    fireEvent.click(button)
    const secondButton = await getByTestId("button-second")
    expect(secondButton).toBeTruthy()
    fireEvent.click(secondButton)
  })

  test("cgetSnapshot", async () => {
    const { asFragment } = render(<AddTicket />)
    expect(asFragment()).toMatchSnapshot()
  })
})
