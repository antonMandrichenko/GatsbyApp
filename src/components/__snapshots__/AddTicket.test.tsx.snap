// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTicket /> cgetSnapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-229 MuiButton-containedSecondary MuiButton-fullWidth"
      data-testid="button-first"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Create list item
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="makeStyles-alert-230"
    />
  </div>
</DocumentFragment>
`;
